src/services/upper-bar-manager/upper-bar-manager.tsx

```diff
@@ -26,19 +27,21 @@ export class UpperBarManager {
     this.componentsRegistry = new Map<number, IconModel>();
     this.logger = logger;
     this.displayedBarComponentRefs = {};
+    this.iconsOrder = {} as IconsOrder;
     UPPER_BAR_PRESETS.forEach((preset) => (this.displayedBarComponentRefs[preset] = createRef()));
-    this.injectDisplayedBarComponentWrapper(config.pluginsIconsOrder);
+    this.injectDisplayedBarComponentWrapper();
   }
 
   public add(icon: IconDto): number | undefined {
     if (UpperBarManager.validateItem(icon)) {
       const newIcon: IconModel = new IconModel(icon);
       this.componentsRegistry.set(newIcon.id, newIcon);
+      this.iconsOrder[icon.displayName] = icon.order;
       newIcon.presets.forEach((preset) => this.displayedBarComponentRefs[preset].current?.update());
-      this.logger.debug(`Icon Id: '${newIcon.id}' '${newIcon.label}' added`);
+      this.logger.debug(`control '${newIcon.displayName}' added, id: '${newIcon.id}' `);
       return newIcon.id;
     }
-    this.logger.warn('Invalid Icon parameters', icon);
+    this.logger.warn('Invalid parameters', JSON.stringify(icon));
```

Comment:
better specify what parameters.
something like "icon cannot be added due to invalid parameters" would be clearer i think